pkg/cli/haproxy.go

```diff
@@ -75,7 +89,31 @@ func runGenHAProxyCmd(cmd *cobra.Command, args []string) error {
 		w = f
 	}
 
-	err = configTemplate.Execute(w, nodeStatuses.Nodes)
+	fs := flag.NewFlagSet("haproxy", flag.ContinueOnError)
```

Comment:
This logic could use a unit test

Reply:
Done, thanks for the suggestion. It caught a bug regarding not reinitializing state between loops.